common.go

```diff
@@ -206,15 +173,9 @@ func (c *Client) DoRequest(ctx context.Context, method, urlStr string, data []by
 	// set any headers provided in context
 	if header, ok := ctx.Value("http.Header").(http.Header); ok {
 		for key, vals := range map[string][]string(header) {
-			if len(vals) >= 1 {
-				// replace existing headers, default, or from Client.headers
-				req.Header.Set(key, vals[0])
-			}
-			if len(vals) > 2 {
```

Comment:
Simplifying fixes the off-by-one error that was here, where the second header wouldn't be applied to the request, but the first and third etc would.